solana/solana-ibc/programs/solana-ibc/src/lib.rs

```diff
@@ -505,6 +525,47 @@ pub struct SendPacket<'info> {
     system_program: Program<'info, System>,
 }
 
+#[derive(Accounts)]
+#[instruction(port_id: ibc::PortId, channel_id_on_b: ibc::ChannelId, base_denom: String)]
+pub struct SendTransfer<'info> {
+    #[account(mut)]
+    sender: Signer<'info>,
+
+    receiver: Option<AccountInfo<'info>>,
+
+    /// The account holding private IBC storage.
+    #[account(mut,seeds = [SOLANA_IBC_STORAGE_SEED],
+              bump)]
+    storage: Account<'info, storage::PrivateStorage>,
+
+    /// The account holding provable IBC storage, i.e. the trie.
+    ///
+    /// CHECK: Account’s owner is checked by [`storage::get_provable_from`]
+    /// function.
+    #[account(mut, seeds = [TRIE_SEED],
+              bump)]
+    trie: UncheckedAccount<'info>,
+
+    /// The guest blockchain data.
+    #[account(mut, seeds = [CHAIN_SEED], bump)]
+    chain: Box<Account<'info, chain::ChainData>>,
+    #[account(mut, seeds = [MINT_ESCROW_SEED], bump)]
+    /// CHECK:
+    mint_authority: Option<UncheckedAccount<'info>>,
+    #[account(mut)]
+    token_mint: Option<Box<Account<'info, Mint>>>,
+    #[account(init_if_needed, payer = sender, seeds = [
+        port_id.as_bytes(), channel_id_on_b.as_bytes(), base_denom[..18].as_bytes(), base_denom[18..].as_bytes()
```

Comment:
Why is base denom twice? Why only first 18 bytes? What if byse denom is shorter than 18 bytes?

Reply:
when we send the token from solana, the denom would be the token mint which is always 32 bytes. And on the seeds, it has a limit on the length because when i tried to add `base_denom` without splitting, it was not able to derive the PDA. I wanted to reduce the length so i took an arbitrary value of 18. But we can use 16 since thats the half of the denom.

Reply:
Got it. It shouldn’t matter since seeds are concatenated but perhaps let’s split at 32 and add a comment describing why we’re doing it?

Reply:
sure sure ill add